tests/specification_tests/end_user_domain/test_history_matching.py

```diff
@@ -0,0 +1,87 @@
+"""
+Use case tests for the domain of history matching. The user for this case
+is a end-user that is trying to perform history matching.
+"""
+
+from argparse import ArgumentParser
+
+import numpy as np
+import pytest
+
+from ert import LibresFacade
+from ert.__main__ import ert_parser
+from ert._c_wrappers.enkf import EnKFMain, ResConfig
+from ert.cli import ENSEMBLE_SMOOTHER_MODE
+from ert.cli.main import run_cli
+from ert.shared.hook_implementations.workflows.disable_parameters import (
+    DisableParametersUpdate,
+)
+
+
+@pytest.mark.integration_test
+def test_that_posterior_has_lower_variance_than_prior(copy_case):
+    # AS A history matcher
+
+    # GIVEN that I am running the polynomial example
+    copy_case("poly_example")
+
+    # WHEN I run ert with ensemble smoother
+    parser = ArgumentParser(prog="test_main")
+    parsed = ert_parser(
+        parser,
+        [
+            ENSEMBLE_SMOOTHER_MODE,
+            "--current-case",
+            "default",
+            "--target-case",
+            "target",
+            "poly.ert",
+            "--port-range",
+            "1024-65535",
+        ],
+    )
+    run_cli(parsed)
+
+    # THEN I expect the variance to reduce with each step
```

Comment:
This is testing "generalized variance", not individual variance elements. Individual variance elements are in fact not guaranteed to decrease after an update (due to existing or induced dependence), but the generalized variance is.

Reply:
Mentioning it, because intuitively is is easy to think that individual variance elements should decrease, and the comment reinforces this (unfortunately wrong) intuition. The code is correct.

Reply:
So the attempt to add documentation revealed this misunderstanding, which I had no idea about reading it the first time.